src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/DefaultSemanticTokensColorsProvider.java

```diff
@@ -29,6 +29,18 @@ public class DefaultSemanticTokensColorsProvider implements SemanticTokensColors
     public @Nullable TextAttributesKey getTextAttributesKey(@NotNull String tokenType,
                                                             @NotNull List<String> tokenModifiers,
                                                             @NotNull PsiFile file) {
+        LSPSemanticTokenType lspSemanticTokenType = LSPSemanticTokenTypes.valueOf(tokenType);
+        if (lspSemanticTokenType != null) {

```

Comment:
Minimalist support for custom semantic token types where the `inheritFrom`  and `textAttributesKey` information is used if available.